packages/common/editors/EditorPortal.js

```diff
@@ -2,24 +2,36 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 
-const editorRoot = document.body;
-
 export default class EditorPortal extends React.Component {
   static propTypes = {
-    children: PropTypes.node
+    children: PropTypes.node.isRequired,
+    target: PropTypes.instanceOf(Element).isRequired
+  };
+
+  // Keep track of when the modal element is added to the DOM
+  state = {
+    isMounted: false
   };
 
   el = document.createElement('div');
 
   componentDidMount() {
-    editorRoot.appendChild(this.el);
+    this.props.target.appendChild(this.el);
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({ isMounted: true });
   }
 
   componentWillUnmount() {
-    editorRoot.removeChild(this.el);
+    this.props.target.removeChild(this.el);
   }
 
   render() {
+    // Don't render the portal until the component has mounted,
```

Comment:
This is needed for editors that require the editor container to be attached to DOM during `componentDidMount`. For e.g legacy editors using jquery to initialize the component in `componentDidMount`